shared/src/balancer/pool_init.rs

```diff
@@ -0,0 +1,256 @@
+//! Balancer pool registry initialization.
+//!
+//! This module contains a component used to initialize Balancer pool registries
+//! with existing data in order to reduce the "cold start" time of the service.
+
+use crate::balancer::{
+    graph_api::{BalancerSubgraphClient, RegisteredWeightedPools},
+    pool_storage::RegisteredWeightedPool,
+};
+use anyhow::{anyhow, Result};
+use contracts::{BalancerV2WeightedPool2TokensFactory, BalancerV2WeightedPoolFactory};
+use ethcontract::{Artifact, H160};
+
+#[derive(Debug, Default, PartialEq)]
+pub struct BalancerRegisteredPools {
+    pub weighted_pools: Vec<RegisteredWeightedPool>,
+    pub weighted_2token_pools: Vec<RegisteredWeightedPool>,
+}
```

Comment:
I assume this is only a temporary struct to be parsed and converted into `PoolStorage` when this is hooked up to the rest.